### ISSUE TITLE
Dont remove bot until current operation finish

### ISSUE THREAD

**Issue by indrif** [Etimo/diamonds2#80]
If a bot is currently performing an operation on the board it should not be removed during the operation. Perhaps this could also be a problem during another bot's operation as well?

Should delete operation also be performed using the operation queue?

**Comment by indrif** [comment]
Any update here @R0tenur ?

**Comment by R0tenur** [comment]
Anyone with time can grab this

**Comment by indrif** [comment]
Branch?

**Comment by R0tenur** [comment]
You see the branch in the related commit above.
I've merged the `operation queue board` with the actual board to be able to enqueue events internaly.
To be able to add private calls to the queue I had to remove the event types and replace them with plain functions.

A have added two helper functions, one for adding last in queue and one for adding first in the queue.

I've added the `remove bot` on an event that is added first in queue.

**Whats left?**
- Two test that are related to removal of the bots fails. I've not spent that much time on how to test the queue.
- I would like to do the calls `move` and `join` private (or remove the queue ones and make these calls to add the action on the queue).
